feat(tray): Add runtime config option to enable/disable system tray

Kishi85 opened the pull request:
<!---
    NOTE: DO NOT DELETE ANY COMMENTS OR SECTIONS OF THIS TEMPLATE.
    THEY ARE IMPORTANT FOR THE MAINTAINERS.
    IT IS OKAY TO LEAVE SECTIONS EMPTY AND BOXES UNCHECKED IF THEY DO NOT APPLY TO YOUR PR.
--->

## Description
<!--- Please include a summary of the changes. --->
Add a runtime config option to enable/disable system tray (and desktop notifications associated with it) so Users can disable the tray icon if they wish to without having to recompile Sunshine. This is especially helpful if there are issues related to the system try (like #4199).


### Screenshot
<!--- Include screenshots if the changes are UI-related. --->


### Issues Fixed or Closed
<!--- If this PR fixes or closes any issues, please list them here. --->
<!--- Close issue example: `- Closes #1` --->
<!--- Fix bug issue example: `- Fixes #2` --->
<!--- Resolve issue example: `- Resolves #3` --->
- Resolves https://github.com/orgs/LizardByte/discussions/334


#### Roadmap Issues
<!--- If this PR solves any roadmap issues (https://github.com/LizardByte/roadmap/issues), please list them here. --->
<!--- e.g. `- Closes https://github.com/LizardByte/roadmap/issues/1` --->


## Type of Change
<!--- Select the type of change your PR introduces. You can select multiple types. --->
- [X] **feat**: New feature (non-breaking change which adds functionality)
- [ ] **fix**: Bug fix (non-breaking change which fixes an issue)
- [ ] **docs**: Documentation only changes
- [ ] **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semicolons, etc.)
- [ ] **refactor**: Code change that neither fixes a bug nor adds a feature
- [ ] **perf**: Code change that improves performance
- [ ] **test**: Adding missing tests or correcting existing tests
- [ ] **build**: Changes that affect the build system or external dependencies
- [ ] **ci**: Changes to CI configuration files and scripts
- [ ] **chore**: Other changes that don't modify src or test files
- [ ] **revert**: Reverts a previous commit
- [ ] **BREAKING CHANGE**: Introduces a breaking change (can be combined with any type above)


## Checklist
<!--- Please check all applicable boxes. If a box does not apply, leave it unchecked. --->
- [X] Code follows the style guidelines of this project
- [X] Code has been self-reviewed
- [X] Code has been commented, particularly in hard-to-understand areas
- [ ] Code docstring/documentation-blocks for new or existing methods/components have been added or updated
- [ ] Unit tests have been added or updated for any new or modified functionality

### AI Usage
<!--- Select the option that best describes AI usage in this PR (choose only one). --->
- [X] **None**: No AI tools were used in creating this PR
- [ ] **Light**: AI provided minor assistance (formatting, simple suggestions)
- [ ] **Moderate**: AI helped with code generation or debugging specific parts
- [ ] **Heavy**: AI generated most or all of the code changes

Kishi85 commented:
> Thank you for another PR!
> 
> This looks pretty good to me. There's a few places that need additional updates when new config options are added.
> 
> * docs: docs/configuration.md
> * config ui: src_assets/common/assets/web/config.html
> 
> The option should be in the same order in all places.

I've made the changes as requested. Thanks for clarifying what I've missed!

This made me realize that I've also missed the config ui point for the other option I've added so I'll put up another PR with a fix for that in a minute. Also it might be a good idea to add some kind of unit test for config options (maybe just grepping for the option in all relevant files would be enough for a start) so this gets caught early.

Kishi85 commented:
I've rebased it to current master. Let's see if that fixes it.

ReenigneArcher commented on src_assets/common/assets/web/configs/tabs/General.vue:
```diff
@@ -133,6 +133,14 @@ function removeCmd(index) {
     ></Checkbox>
+
+    <!-- Enable system tray -->
+    <Checkbox class="mb-3"
```
This should not be shown for macOS.

ReenigneArcher reviewed (commented):
I thought of one more thing, since the tray doesn't work on macOS yet.

I'm trying to change that in https://github.com/LizardByte/Sunshine/pull/3818 but there is one bug that needs to be fixed before that can be merged.

ReenigneArcher commented:
Apologies, but I ended up wrapping up https://github.com/LizardByte/Sunshine/pull/3818 last night.

This will require fixing a few conflicts and the tray now works on macOS as well.

Kishi85 commented:
> Apologies, but I ended up wrapping up #3818 last night.
> 
> This will require fixing a few conflicts and the tray now works on macOS as well.

I've rebased to the new master and squashed my changes in the process to simplify adapting this to #3818. 

I moved the config value check from system_tray.cpp to main.cpp right where the system tray is initially started. I'm not sure if this is the best way to do it though. I've also removed the commit with the MacOS restriction.
